forbidigo/patterns.go

```diff
@@ -0,0 +1,43 @@
+package forbidigo
+
+import (
+	"fmt"
+	"regexp"
+	"regexp/syntax"
+	"strings"
+)
+
+type pattern struct {
+	pattern *regexp.Regexp
+	msg     string
+}
+
+func parse(ptrn string) (*pattern, error) {
+	p := &pattern{}
+	var err error
+	p.pattern, err = regexp.Compile(ptrn)
+	if err != nil {
+		return nil, fmt.Errorf("unable to compile pattern `%s`: %s", ptrn, err)
+	}
+	re, err := syntax.Parse(ptrn, syntax.Perl)
```

Comment:
I couldn't spot a way in the stdlib to convert between a `regexp.Regexp` and a `syntax.Regexp`, so I think we might have to parse it twice, which is a bit of a shame.

Reply:
I agree it's not ideal but it only happens once per expression so I'm not too concerned.